docs/interfaces.rst

```diff
@@ -24,23 +24,29 @@ The defined interface can then be used to make external calls, given a contract
 
     @external
     def test(foobar: FooBar):
-        foobar.calculate()
+        extcall foobar.test1()
+
+    @external
+    def test2(foobar: FooBar) -> uint256:
+        return staticcall foobar.calculate()
 
 The interface name can also be used as a type annotation for storage variables. You then assign an address value to the variable to access that interface. Note that casting an address to an interface is possible, e.g. ``FooBar(<address_var>)``:
 
 .. code-block:: vyper
 
     foobar_contract: FooBar
 
-    @external
+    @deploy
     def __init__(foobar_address: address):
         self.foobar_contract = FooBar(foobar_address)
 
     @external
     def test():
-        self.foobar_contract.calculate()
+        extcall self.foobar_contract.test1()
+
+Specifying ``payable`` or ``nonpayable`` annotation in the interface indicates that the call made to the external contract will be able to alter storage, whereas ``view`` and ``pure`` calls will use a ``STATICCALL`` ensuring no storage can be altered during execution. Additionally, ``payable`` allows non-zero value to be sent along with the call.
 
-Specifying ``payable`` or ``nonpayable`` annotation indicates that the call made to the external contract will be able to alter storage, whereas the ``view`` ``pure`` call will use a ``STATICCALL`` ensuring no storage can be altered during execution. Additionally, ``payable`` allows non-zero value to be sent along with the call.
+The ``extcall`` or ``staticcall`` keyword is required to precede the external call to distinguish it from internal calls. The keyword must match the visibility of the function, ``staticcall`` for ``pure`` and ``view`` functions, and ``extcall`` for ``payable`` and ``nonpayable`` functions. Additionally, the output of a ``staticcall`` must be assigned to a result.
```

Comment:
```suggestion
Either the ``extcall`` or ``staticcall`` keyword is required to precede the external call to distinguish it from internal calls. The keyword must match the visibility of the function, ``staticcall`` for ``pure`` and ``view`` functions, and ``extcall`` for ``payable`` and ``nonpayable`` functions. Additionally, the output of a ``staticcall`` must be assigned to a result.
```

Maybe it's nice to explain what happens if you specify `(non)payable` incorrectly

Reply:
do you mean `extcall` vs `staticcall`?

Reply:
efae8ad776663c7b93e1065316c1e6e31869bece